packages/vscode-extension/src/plugins/network/network-plugin.ts

```diff
@@ -1,18 +1,54 @@
 import http, { Server } from "http";
+import { spawn } from "child_process";
+import path from "path";
 import { commands, Disposable, window } from "vscode";
 import { WebSocketServer, WebSocket } from "ws";
 import { Devtools } from "../../project/devtools";
 import { ToolKey, ToolPlugin } from "../../project/tools";
 import { extensionContext } from "../../utilities/extensionContext";
+
+import { Logger } from "../../Logger";
 import { NetworkDevtoolsWebviewProvider } from "./NetworkDevtoolsWebviewProvider";
 
 export const NETWORK_PLUGIN_ID = "network";
 
+function startViteServer(): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const process = spawn("npm", ["run", "watch:network-webview"], {
+      cwd: path.join(__dirname, ".."),
+      shell: true,
+    });
+
+    process.stdout.on("data", (data) => {
+      const output = data.toString();
+
+      if (output.includes("ready in") || output.includes("Local:")) {
+        resolve();
+      }
+    });
+
+    process.stderr.on("data", (data) => {
+      Logger.error("ERROR:", data.toString());
+    });
+
+    process.on("close", (code) => {
+      if (code !== 0) {
+        reject(new Error(`Process exited with code ${code}`));
+      } else {
+        Logger.debug(`Process exited with code ${code}`);
+      }
+    });
+  });
+}
+
 let initialzed = false;
```

Comment:
```suggestion
let initialized = false;
```
while we are here we can also fix a typo